internal/constants/scripts.go

```diff
@@ -66,140 +70,149 @@ function map_host_macs_to_interfaces() {
   for nic in $( ls $SYS_CLASS_NET_DIR )
   do
     mac=$(cat $SYS_CLASS_NET_DIR/$nic/address | tr '[:lower:]' '[:upper:]')
-    host_macs_to_hw_iface[$mac]=$nic
+    [[ -n "$mac" ]] && host_macs_to_hw_iface[$mac]=$nic
   done
 }
 
-function find_host_directory_by_mac_address() {
-  for d in $(ls -d ${NMCONNECTIONS_DIR}/host*)
-  do
+# Copy all nmstate files representing a host nic
+function copy_physical_nics_files() {
+  host_src_dir=$1
+  mapping_file=$2
+  host_work_dir=$3
+  for mac_address in $(cat $mapping_file | cut -d= -f1 | tr '[:lower:]' '[:upper:]');  do
+    host_nic="${host_macs_to_hw_iface[${mac_address}]:-}"
+    if [[  -n "$host_nic" ]]; then
+      logical_nic="${logical_mac_nic_map[${mac_address}]:-}"
+      if [[ -n "$(ls -A $host_src_dir/*.nmconnection )" ]] ; then
+        for nmfile in $host_src_dir/*.nmconnection ; do
+          filename=$(basename $nmfile)
+          prefix="${filename%%.*}"
+          extension="${filename#*.}"
+          [[ -n "$prefix" ]] && [[ "$prefix" == "$logical_nic" ]] &&  cp $nmfile $host_work_dir/${host_nic}.${extension} && rm -f $nmfile
```

Comment:
Would it be easier for this to be a move rather than copy and delete?

Reply:
OK